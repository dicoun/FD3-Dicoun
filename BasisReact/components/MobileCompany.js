import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';

import './MobileCompany.css';

import {voteEvents} from './events';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
        otch: PropTypes.string.isRequired,
        balance: PropTypes.string.isRequired,
        state: PropTypes.string.isRequired,
        disabledFlag: PropTypes.bool.isRequired
      })
    ),
  };

  state = {
    clients: this.props.clients
  };

  filterFlag = '';
  
  filterAll = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.forEach( (c,i) => {
        if(c.renderFlag == false){
          let newClient={...c, renderFlag:true}; // копия хэша изменившегося клиента
          newClients[i]=newClient;
        }
      });
      this.filterFlag = 'all';
      this.setState({
        clients: newClients
      });
    }
  };

  filterActive = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.forEach( (c,i) => {
        if((c.state == 'active') && (c.renderFlag == false)){
          let newClient={...c, renderFlag:true}; // копия хэша изменившегося клиента
          newClients[i]=newClient;
        }
      });
      this.filterFlag = 'active';
      this.setState({
        clients: newClients
      });
    }
  };

  filterBlocked = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.forEach( (c,i) => {
        if((c.state == 'blocked') && (c.renderFlag == false)){
          let newClient={...c, renderFlag:true}; // копия хэша изменившегося клиента
          newClients[i]=newClient;
        }
      });
      this.filterFlag = 'blocked';
      this.setState({
        clients: newClients
      });
    }
  };

  setNewSurnameRef = (ref) => {
    this.newSurnameRef=ref;
  };

  setNewBallanceRef = (ref) => {
    this.newBallanceRef=ref;
  };

  setNewNameRef = (ref) => {
    this.newNameRef=ref;
  };

  setNewMiddleNameRef = (ref) => {
    this.newMiddleNameRef=ref;
  };

  AddBtn = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.push({
        id: newClients.length + 1,
        fam: '',
        im: '',
        otch: '',
        balance: '',
        state: 'active',
        disabledFlag: false,
        addFlag: true
      });
      this.setState({
        clients: newClients
      });
    }
  }

  EditBtn = (id) => {
    let changed=false;
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c}; // копия хэша изменившегося клиента
      if(c.id == id){
        newClient.disabledFlag=false;
        newClients[i]=newClient;
        changed=true;
      }
    });
    if ( changed ){
      this.setState({clients: newClients});
    }
  }

  DeleteBtn = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    let delIndex = 0;
    newClients.forEach( (c,i) => {
      if(c.id == id){
        delIndex = i;
      }
    });
    newClients.splice(delIndex,1);
    this.setState({clients: newClients});
  }

  SaveBtn = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c}; // копия хэша изменившегося клиента
      if(c.id == id){
        if ( this.newSurnameRef ) { // всегда проверяем - мало ли метод вызовется когда DOM-элемента уже нет или ещё нет?
          let newSurname=this.newSurnameRef.value;
          let newBallance=this.newBallanceRef.value;
          newClient.disabledFlag=true;
          newClient.fam=newSurname;
          newClient.balance=newBallance;
          if(c.addFlag){
            let newName=this.newNameRef.value;
            newClient.im=newName;
            let newMiddleName=this.newMiddleNameRef.value;
            newClient.otch=newMiddleName;
            newClient.addFlag=false;
          }
          newClients[i]=newClient;
        }
      }
    });
    this.setState({clients: newClients});
  }

  CancelBtn = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c}; // копия хэша изменившегося клиента
      if(c.id == id){
        newClient.disabledFlag=true;
        if(c.addFlag){
          newClient.addFlag=false;
        }
        newClients[i]=newClient; 
      }
    });
    this.setState({clients: newClients});
  }

  componentDidMount = () => {
    voteEvents.addListener('EditBtnClicked',this.EditBtn);
    voteEvents.addListener('DeleteBtnClicked',this.DeleteBtn);
    voteEvents.addListener('SaveBtnClicked',this.SaveBtn);
    voteEvents.addListener('CancelBtnClicked',this.CancelBtn);
    voteEvents.addListener('setSurnameRef',this.setNewSurnameRef);
    voteEvents.addListener('setBallanceRef',this.setNewBallanceRef);
    voteEvents.addListener('setNameRef',this.setNewNameRef);
    voteEvents.addListener('setMiddleNameRef',this.setNewMiddleNameRef);
  };

  componentWillUnmount = () => {
    voteEvents.removeListener('EditBtnClicked',this.EditBtn);
    voteEvents.removeListener('DeleteBtnClicked',this.DeleteBtn);
    voteEvents.removeListener('SaveBtnClicked',this.SaveBtn);
    voteEvents.removeListener('CancelBtnClicked',this.CancelBtn);
    voteEvents.removeListener('setSurnameRef',this.setNewSurnameRef);
    voteEvents.removeListener('setBallanceRef',this.setNewBallanceRef);
    voteEvents.removeListener('setNameRef',this.setNewNameRef);
    voteEvents.removeListener('setMiddleNameRef',this.setNewMiddleNameRef);
  };
  
  render() {
    console.log("MobileCompany render");
    var clientsCode = [];
    if(this.props.clients){
      clientsCode=this.state.clients.map( client => {
        if(client.state == this.filterFlag){
          return <MobileClient key={client.id} client={client}/>;
        }
        if(this.filterFlag == 'all' || this.filterFlag == ''){
          return <MobileClient key={client.id} client={client}/>;
        }
      });
    }
    return (
      <div className='MobileCompany'>
        <input type="button" kind="All" value="Все" onClick={this.filterAll} />
        <input type="button" kind="Active" value="Активные" onClick={this.filterActive} />
        <input type="button" kind="Blocked" value="Заблокированные" onClick={this.filterBlocked} />
        <hr></hr>
        <div className='MobileCompanyClients'>
          <table className='MobileCompanyTable'>
            <tbody>
              <tr>
                <th>{'Фамилия'}</th>
                <th>{'Имя'}</th>
                <th>{'Отчество'}</th>
                <th>{'Баланс'}</th>
                <th>{'Статус'}</th>
                <th>{'Редактировать'}</th>
                <th>{'Удалить'}</th>
              </tr>
              {clientsCode}
            </tbody>
          </table>
        </div>
        <hr></hr>
        <input type="button" kind="addClient" value="Добавить клиента" onClick={this.AddBtn} />
      </div>
    );
  }
}

export default MobileCompany;
