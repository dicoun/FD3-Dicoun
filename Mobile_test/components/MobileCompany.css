table {
  font-family: "Lucida Sans Unicode", "Lucida Grande", Sans-Serif;
  font-size: 14px;
  background: white;
  max-width: 100%;
  width: 100%;
  border-collapse: collapse;
  text-align: left;
}
th {
  font-weight: bold;
  color: black;
  border-bottom: 1px solid grey;
  padding: 10px 8px;
}
td {
  color:black;
  padding: 9px 8px;
  border-bottom: 1px solid grey;
  transition: .3s linear;
}
.tabName { 
  display:inline-block; 
  margin-left: 20px; 
}
.tabBallance { 
  display:inline-block; 
  margin-left: 26px; 
}

