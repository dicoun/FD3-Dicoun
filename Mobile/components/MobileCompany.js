import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';

import './MobileCompany.css';

import {voteEvents} from './events';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
        otch: PropTypes.string.isRequired,
        balance: PropTypes.string.isRequired,
        state: PropTypes.string.isRequired,
        disabledFlag: PropTypes.bool.isRequired
      })
    ),
  };

  state = {
    name: this.props.name,
    clients: this.props.clients
  };

  setName1 = () => {
    this.setState({name:'МТС'});
  };

  setName2 = () => {
    this.setState({name:'Velcom'});
  };
  
  filterAll = () => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c, filterFlag:'all'}; // копия хэша изменившегося клиента
      newClients[i]=newClient;
    });
    this.setState({
      clients: newClients
    });
  };

  filterActive = () => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c, filterFlag:'active'}; // копия хэша изменившегося клиента
      newClients[i]=newClient;
    });
    this.setState({
      clients: newClients
    });
  };

  filterBlocked = () => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c, filterFlag:'blocked'}; // копия хэша изменившегося клиента
      newClients[i]=newClient;
    });
    this.setState({
      clients: newClients
    });
  };

  /*setNewSurnameRef = (ref) => {
    this.newSurnameRef=ref;
  };

  setNewBallanceRef = (ref) => {
    this.newBallanceRef=ref;
  };*/

  AddBtn = () => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.push({
      id: newClients.length + 1,
      fam: '',
      im: '',
      otch: '',
      balance: '',
      state: 'active',
      disabledFlag: false,
      addFlag: true
    });
    this.setState({
      clients: newClients
    });
  }

  EditBtn = (id) => {
    let changed=false;
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c}; // копия хэша изменившегося клиента
      if(c.id == id){
        newClient.disabledFlag=false;
        newClients[i]=newClient;
        changed=true;
      }
    });
    if ( changed ){
      this.setState({clients: newClients});
    }
  }

  DeleteBtn = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    let delIndex = 0;
    newClients.forEach( (c,i) => {
      if(c.id == id){
        delIndex = i;
      }
    });
    newClients.splice(delIndex,1);
    this.setState({clients: newClients});
  }

  componentDidMount = () => {
    voteEvents.addListener('EditBtnClicked',this.EditBtn);
    voteEvents.addListener('DeleteBtnClicked',this.DeleteBtn);
    /*voteEvents.addListener('setSurnameRef',this.setNewSurnameRef);
    voteEvents.addListener('setBallanceRef',this.setNewBallanceRef);*/
  };

  componentWillUnmount = () => {
    voteEvents.removeListener('EditBtnClicked',this.EditBtn);
    voteEvents.removeListener('DeleteBtnClicked',this.DeleteBtn);
    /*voteEvents.removeListener('setSurnameRef',this.setNewSurnameRef);
    voteEvents.removeListener('setBallanceRef',this.setNewBallanceRef);*/
  };
  
  render() {
    console.log("MobileCompany render");
    var clientsCode=this.state.clients.map( client => {
        if(client.state == client.filterFlag){
          return <MobileClient key={client.id} client={client}/>;
        }
        if(client.filterFlag == 'all' || client.filterFlag == undefined){
          return <MobileClient key={client.id} client={client}/>;
        }
      }
    );
    return (
      <div className='MobileCompany'>
        <input type="button" value="Velcom" onClick={this.setName2} />
        <input type="button" value="МТС" onClick={this.setName1} />
        <div className='MobileCompanyName'>Компания: {this.state.name}</div>
        <hr></hr>
        <input type="button" value="Все" onClick={this.filterAll} />
        <input type="button" value="Активные" onClick={this.filterActive} />
        <input type="button" value="Заблокированные" onClick={this.filterBlocked} />
        <hr></hr>
        <div className='MobileCompanyClients'>
          <table className='MobileCompanyTable'>
            <tbody>
              <tr>
                <th>{'Фамилия'}</th>
                <th>{'Имя'}</th>
                <th>{'Отчество'}</th>
                <th>{'Баланс'}</th>
                <th>{'Статус'}</th>
                <th>{'Редактировать'}</th>
                <th>{'Удалить'}</th>
              </tr>
              {clientsCode}
            </tbody>
          </table>
        </div>
        <hr></hr>
        <input type="button" value="Добавить клиента" onClick={this.AddBtn} />
      </div>
    );
  }
}

export default MobileCompany;
