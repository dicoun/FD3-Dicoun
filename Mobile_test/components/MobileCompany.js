import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';

import './MobileCompany.css';

import {voteEvents} from './events';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
        otch: PropTypes.string.isRequired,
        balance: PropTypes.string.isRequired,
        state: PropTypes.string.isRequired,
        disabledFlag: PropTypes.bool.isRequired
      })
    ),
  };

  state = {
    clients: this.props.clients
  };

  filterFlag = '';
  newNameRef = [];
  newSurnameRef = [];
  newBallanceRef = [];
  newMiddleNameRef = [];

  
  filterAll = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.forEach( (c,i) => {
        if(c.renderFlag == false){
          let newClient={...c, renderFlag:true}; // копия хэша изменившегося клиента
          newClients[i]=newClient;
        }
      });
      this.filterFlag = 'all';
      this.setState({
        clients: newClients
      });
    }
  };

  filterActive = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.forEach( (c,i) => {
        if((c.state == 'active') && (c.renderFlag == false)){
          let newClient={...c, renderFlag:true}; // копия хэша изменившегося клиента
          newClients[i]=newClient;
        }
      });
      this.filterFlag = 'active';
      this.setState({
        clients: newClients
      });
    }
  };

  filterBlocked = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.forEach( (c,i) => {
        if((c.state == 'blocked') && (c.renderFlag == false)){
          let newClient={...c, renderFlag:true}; // копия хэша изменившегося клиента
          newClients[i]=newClient;
        }
      });
      this.filterFlag = 'blocked';
      this.setState({
        clients: newClients
      });
    }
  };

  setNewSurnameRef = (ref, id) => {
    this.newSurnameRef.push({ref:ref,id:id});
  };

  setNewBallanceRef = (ref, id) => {
    this.newBallanceRef.push({ref:ref,id:id});
  };

  setNewNameRef = (ref, id) => {
    this.newNameRef.push({ref:ref,id:id});
  };

  setNewMiddleNameRef = (ref, id) => {
    this.newMiddleNameRef.push({ref:ref,id:id});;
  };

  AddBtn = () => {
    if(this.state.clients){
      let newClients=[...this.state.clients]; // копия самого массива клиентов
      newClients.push({
        id: newClients.length + 1,
        fam: '',
        im: '',
        otch: '',
        balance: '',
        state: 'active',
        disabledFlag: false,
        addFlag: true
      });
      this.setState({
        clients: newClients
      });
    }
  }

  EditBtn = (id) => {
    let changed=false;
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c}; // копия хэша изменившегося клиента
      if(c.id == id){
        newClient.disabledFlag=false;
        newClients[i]=newClient;
        changed=true;
      }
    });
    if ( changed ){
      this.setState({clients: newClients});
    }
  }

  DeleteBtn = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    let delIndex = 0;
    newClients.forEach( (c,i) => {
      if(c.id == id){
        delIndex = i;
      }
    });
    newClients.splice(delIndex,1);
    this.setState({clients: newClients});
  }

  SaveBtn = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c}; // копия хэша изменившегося клиента
      if(c.id == id){
        if ( this.newSurnameRef.length ) { // всегда проверяем - мало ли метод вызовется когда DOM-элемента уже нет или ещё нет?
          this.newSurnameRef.forEach((rec,i) => {
            if(rec.id == id){
              if(rec.ref){
                let newSurname=rec.ref.value;
                newClient.fam=newSurname;
              }
            }
          });
        }
        if(this.newBallanceRef.length){
          this.newBallanceRef.forEach((rec,i) => {
            if(rec.id == id){
              if(rec.ref){
                let newBallance=rec.ref.value;
                newClient.balance=newBallance;
              }
            }
          });
        }
        newClient.disabledFlag=true;
        if(c.addFlag){
          if(this.newNameRef.length){
            this.newNameRef.forEach((rec,i) => {
              if(rec.id == id){
                if(rec.ref){
                  let newName=rec.ref.value;
                  newClient.im=newName;
                }
              }
            });
          }
          if(this.newMiddleNameRef.length){
            this.newMiddleNameRef.forEach((rec,i) => {
              if(rec.id == id){
                if(rec.ref){
                  let newMiddleName=rec.ref.value;
                  newClient.otch=newMiddleName;
                }
              } 
            });
          }
          newClient.addFlag=false;
        }
        newClients[i]=newClient;
      }
    });
    this.newSurnameRef = [];
    this.newBallanceRef = [];
    this.newNameRef = [];
    this.newMiddleNameRef = [];

    this.setState({clients: newClients});
  }

  CancelBtn = (id) => {
    let newClients=[...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      let newClient={...c}; // копия хэша изменившегося клиента
      if(c.id == id){
        newClient.disabledFlag=true;
        if(c.addFlag){
          newClient.addFlag=false;
        }
        newClients[i]=newClient; 
      }
    });
    this.setState({clients: newClients});
  }

  componentDidMount = () => {
    voteEvents.addListener('EditBtnClicked',this.EditBtn);
    voteEvents.addListener('DeleteBtnClicked',this.DeleteBtn);
    voteEvents.addListener('SaveBtnClicked',this.SaveBtn);
    voteEvents.addListener('CancelBtnClicked',this.CancelBtn);
    voteEvents.addListener('setSurnameRef',this.setNewSurnameRef);
    voteEvents.addListener('setBallanceRef',this.setNewBallanceRef);
    voteEvents.addListener('setNameRef',this.setNewNameRef);
    voteEvents.addListener('setMiddleNameRef',this.setNewMiddleNameRef);
  };

  componentWillUnmount = () => {
    voteEvents.removeListener('EditBtnClicked',this.EditBtn);
    voteEvents.removeListener('DeleteBtnClicked',this.DeleteBtn);
    voteEvents.removeListener('SaveBtnClicked',this.SaveBtn);
    voteEvents.removeListener('CancelBtnClicked',this.CancelBtn);
    voteEvents.removeListener('setSurnameRef',this.setNewSurnameRef);
    voteEvents.removeListener('setBallanceRef',this.setNewBallanceRef);
    voteEvents.removeListener('setNameRef',this.setNewNameRef);
    voteEvents.removeListener('setMiddleNameRef',this.setNewMiddleNameRef);
  };
  
  render() {
    console.log("MobileCompany render");
    var clientsCode = [];
    if(this.props.clients){
      clientsCode=this.state.clients.map( client => {
        if(client.state == this.filterFlag){
          return <MobileClient key={client.id} client={client}/>;
        }
        if(this.filterFlag == 'all' || this.filterFlag == ''){
          return <MobileClient key={client.id} client={client}/>;
        }
      });
    }
    return (
      <div className='MobileCompany'>
        <input type="button" kind="All" value="Все" onClick={this.filterAll} />
        <input type="button" kind="Active" value="Активные" onClick={this.filterActive} />
        <input type="button" kind="Blocked" value="Заблокированные" onClick={this.filterBlocked} />
        <hr></hr>
        <div className='MobileCompanyClients'>
          <table className='MobileCompanyTable'>
            <tbody>
              <tr>
                <th>{'Фамилия'}</th>
                <th>{'Имя'}</th>
                <th>{'Отчество'}</th>
                <th>{'Баланс'}</th>
                <th>{'Статус'}</th>
                <th>{'Редактировать'}</th>
                <th>{'Удалить'}</th>
              </tr>
              {clientsCode}
            </tbody>
          </table>
        </div>
        <hr></hr>
        <input type="button" kind="addClient" value="Добавить клиента" onClick={this.AddBtn} />
      </div>
    );
  }
}

export default MobileCompany;
