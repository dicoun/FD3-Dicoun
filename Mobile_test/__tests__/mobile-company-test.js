import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';

test('работа MobileCompany', () => {

  const component = renderer.create(
    <MobileCompany />
  );

  let componentTree=component.toJSON();
  /*expect(componentTree).toMatchSnapshot();*/

  // найдём в вёрстке компонента саму кнопку
  const addBtnElem = component.root.find( el => el.type=='input' && el.props.kind == 'addClient' ); 
  // и "нажмём" на неё
  addBtnElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента саму кнопку
  const allBtnElem = component.root.find( el => el.type=='input' && el.props.kind == 'All' ); 
  // и "нажмём" на неё
  allBtnElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // найдём в вёрстке компонента саму кнопку
  const activeBtnElem = component.root.find( el => el.type=='input' && el.props.kind == 'Active' ); 
  // и "нажмём" на неё
  activeBtnElem.props.onClick();

  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

   // найдём в вёрстке компонента саму кнопку
   const blockedBtnElem = component.root.find( el => el.type=='input' && el.props.kind == 'Blocked' ); 
   // и "нажмём" на неё
   blockedBtnElem.props.onClick();
 
   // получаем уже изменённый снэпшот
   componentTree=component.toJSON();
   expect(componentTree).toMatchSnapshot();
    
});
