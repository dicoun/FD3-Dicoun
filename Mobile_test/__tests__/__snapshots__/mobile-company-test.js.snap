// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileCompany 1`] = `
<div
  className="MobileCompany"
>
  <input
    kind="All"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    kind="Active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    kind="Blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <div
    className="MobileCompanyClients"
  >
    <table
      className="MobileCompanyTable"
    >
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <hr />
  <input
    kind="addClient"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 2`] = `
<div
  className="MobileCompany"
>
  <input
    kind="All"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    kind="Active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    kind="Blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <div
    className="MobileCompanyClients"
  >
    <table
      className="MobileCompanyTable"
    >
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <hr />
  <input
    kind="addClient"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 3`] = `
<div
  className="MobileCompany"
>
  <input
    kind="All"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    kind="Active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    kind="Blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <div
    className="MobileCompanyClients"
  >
    <table
      className="MobileCompanyTable"
    >
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <hr />
  <input
    kind="addClient"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 4`] = `
<div
  className="MobileCompany"
>
  <input
    kind="All"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    kind="Active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    kind="Blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <div
    className="MobileCompanyClients"
  >
    <table
      className="MobileCompanyTable"
    >
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <hr />
  <input
    kind="addClient"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;
